### PR TITLE
MAGN-8477 Uninstalling package does not update the library ui

### PR DESCRIPTION
**Pull request by ramramps**
### Purpose

This PR fixes the issue with UI not updated when a package is uninstalled. This occurs only for those packages that have a different category in their dyf file. (ex: clockwork). 
This PR also fixes the duplicate entries for packages (after uninstall, install the same package, you can see duplicate entries).

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files

### Reviewers

@mjkkirschner

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs [hunk 1]
```diff
@@ -644,8 +646,11 @@ internal void InsertEntry(NodeSearchElementViewModel entry, IEnumerable<string>
         private void AddEntryToExistingCategory(NodeCategoryViewModel category,
             NodeSearchElementViewModel entry)
         {
-            category.RequestBitmapSource += SearchViewModelRequestBitmapSource;
-            category.Entries.Add(entry);
+            category.RequestBitmapSource += SearchViewModelRequestBitmapSource; 
+            //Check if the category exists already. This check is vital if 
+            // the category name is updated. ex : clockwork package
+            if(category.Entries.All(x => x.FullName != entry.FullName))
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs]: can you add braces around this if statement please

File: src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs [hunk 2]
```diff
@@ -644,8 +646,11 @@ internal void InsertEntry(NodeSearchElementViewModel entry, IEnumerable<string>
         private void AddEntryToExistingCategory(NodeCategoryViewModel category,
             NodeSearchElementViewModel entry)
         {
-            category.RequestBitmapSource += SearchViewModelRequestBitmapSource;
-            category.Entries.Add(entry);
+            category.RequestBitmapSource += SearchViewModelRequestBitmapSource; 
+            //Check if the category exists already. This check is vital if 
+            // the category name is updated. ex : clockwork package
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs]: can you make this example a little more detailed, how was the category transformed? x => y? The package author made this change? I think that clarification would be useful to readers of this code.

**Comment by mjkkirschner** [comment]
@ramramps a few small things and also is it possible to add a test for this to make sure it does not regress?

theres also a merge conflict - can you fix that? and I'll check it out again.

**Comment by ramramps** [comment]
@mjkkirschner - ping

**Comment by mjkkirschner** [comment]
@ramramps what about a test?

**Comment by ramramps** [comment]
@mjkkirschner - The tests are already there for this. Here the changes are very specific to  user task - Uninstall and install. I am not sure whether this can be tested.

**Comment by ramramps** [comment]
@mjkkirschner  - pIng!!

**Comment by mjkkirschner** [comment]
@ramramps - thanks, LGTM

**Comment by jnealb** [comment]
@mjkkirschner @ramramps MAGN-8477's Release is 9.0 - this needs to be cherry-picked.

**Pull request merged**